byzcoin/transaction.go

```diff
@@ -90,6 +90,20 @@ func (args Arguments) Names() []string {
 	return names
 }
 
+// Hash calculates the hash over all instructions and their signatures.
+// This creates a unique hash with regard to two sets of instructions
+// that only differ with their signature.
+func (ctx *ClientTransaction) Hash() []byte {
```

Comment:
this function seems to be similar to `DeriveID`, except `DeriveID` does domain separation. can we use that instead?

Reply:
oh nevermind, `DeriveID` is only for one `Instruction`.

my general feeling is that future developers will be confused which `Hash` to use. they might not be aware that there's one under `Instructions` if they see the `ClientTransaction` one first. my suggestion is to move this one to `Instructions` but rename is to `HashWithSignatures`, for example. so that future developers see two versions of it at the same place and then needs to make a conscious decision on which one to use